allennlp/modules/text_field_embedders/basic_text_field_embedder.py

```diff
@@ -107,20 +109,20 @@ def forward(self, text_field_input: Dict[str, torch.Tensor], num_wrapping_dims:
                     # If `indexer_key` is None, we map it to `None`.
                     tensors = [(text_field_input[indexer_key] if indexer_key is not None else None)
                                for indexer_key in indexer_map]
-                    token_vectors = embedder(*tensors)
+                    token_vectors = embedder(*tensors, **kwargs)
```

Comment:
This is not safe.  You can't pass the same `kwargs` to all embedders, because not all of them are going to take the same extra arguments.  I can think of two options to handle this:

1. Do some inspection here to figure out which extra arguments are compatible with the specific `embedder` that you're looking at in the loop, and only pass those.
2. Instead of modifying this class, make a new `Indexer` that's a `MultilingualElmoIndexer` that adds a separate key to the token mapping (that would end up looking like `elmo-lang`, with a string value), like we do with the OpenAI and BERT indexers (e.g., see [here](https://github.com/allenai/allennlp/blob/79e2cf7b677f84daefaf63acc2a8e3833782de23/allennlp/data/token_indexers/openai_transformer_byte_pair_indexer.py#L221-L228) and [here](https://github.com/allenai/allennlp/blob/79e2cf7b677f84daefaf63acc2a8e3833782de23/allennlp/tests/fixtures/openai_transformer/config_small.jsonnet#L24-L27)).  This would require adding a field to the `Token` class that stores the language, so that you know which language you're using.

I'm not sure which option to go with.  Maybe slightly less work to go with option 1, and it seems like it might be potentially useful for other applications, also.  Option 2 makes it so you're not making API changes just to support this, and it doesn't look like much additional code, as you can basically just subclass `ElmoIndexer` and tweak one method.  Except that you have to add a field to `Token`...

Reply:
Will the desired behaviour of this PR be that it will also be token embedder agnostic? At the moment, I believe the only token embedder that was changed was the addition of a new `elmo_token_embedder_multilang.py` file.

If one would like to use the multilingual biaffine parser but without ELMo, e.g. for parsing related languages where we may not have trained ELMo models, are there many changes that would need to be made to other token embedders? From my understanding, in such a case we would also want to make sure the ["embedding" token embedder](https://github.com/TalSchuster/allennlp-MultiLang/blob/7b3c4aa1208218ad71f733767b8150531e99720e/allennlp/modules/token_embedders/embedding.py#L29) would also handle this extra argument.

Thanks for the great work in the PR!

Reply:
Yeah, it's not desirable (or even really meaningful) to change all of the other token embedders to handle this argument.  This is why the existing PR needs to be fixed, as it will currently break if you try to use anything other than the multilingual ELMo embedder.  Either one of the two options above would fix this, I believe.

Reply:
I went with the first suggestion since, if I understand correctly, the second option requires an identifier per token, which is a bit of overhead since we get sentences from the same language